native/src/textdraw.cpp

```diff
@@ -920,32 +920,54 @@ void FontRegistry::drawHbTextOnPath(SkCanvas *canvas, std::string textS, SkPath
 	hb_buffer_destroy(hb_buffer);
 	hb_font_destroy(hb_font);
 
+	SkPathMeasure meas(path, false);
+	// check correlation between harfbuzz and skia
+	if (xy[length - 1].x() > meas.getLength()) 
+	{
+		SkScalar correlation = meas.getLength() / xy[length - 1].x();
+		for (int i = 0; i < length; ++i)
+		{
+			xy[i].set(xy[i].x() * correlation, xy[i].y());
```

Comment:
if correlation = 0.1, does it mean glyphs will go over each other?

Reply:
This is unlikely but I added check correlation less than half of measure lenght.